src/libs/wcautil/WixToolset.WcaUtil/qtexec.cpp

```diff
@@ -30,42 +38,51 @@ static HRESULT CreatePipes(
     HANDLE hInRead = INVALID_HANDLE_VALUE;
     HANDLE hInWrite = INVALID_HANDLE_VALUE;
 
+    // Generate unique pipe names
+    rs = ::UuidCreate(&guid);
+    hr = HRESULT_FROM_RPC(rs);
+    ExitOnFailure(hr, "Failed to create working folder guid.");
+    
+    fRes = ::StringFromGUID2(guid, wzGuid, countof(wzGuid));
+    ExitOnNull(fRes, hr, E_OUTOFMEMORY, "Failed to convert UUID to string");
+    
+    hr = StrAllocFormatted(&szStdInPipeName, L"\\\\.\\pipe\\%ls-stdin", wzGuid);
+    ExitOnFailure(hr, "Failed to create stdin pipe name.");
+    
+    hr = StrAllocFormatted(&szStdOutPipeName, L"\\\\.\\pipe\\%ls-stdout", wzGuid);
+    ExitOnFailure(hr, "Failed to create stdout pipe name.");
+
     // Fill out security structure so we can inherit handles
     ::ZeroMemory(&sa, sizeof(SECURITY_ATTRIBUTES));
     sa.nLength = sizeof(SECURITY_ATTRIBUTES);
     sa.bInheritHandle = TRUE;
     sa.lpSecurityDescriptor = NULL;
-
+    
     // Create pipes
-    if (!::CreatePipe(&hOutTemp, &hOutWrite, &sa, 0))
-    {
-        ExitOnLastError(hr, "Failed to create output pipe");
-    }
+    hOutTemp = ::CreateNamedPipeW(szStdOutPipeName, PIPE_ACCESS_INBOUND | FILE_FLAG_OVERLAPPED, PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_WAIT, 1, OUTPUT_BUFFER, OUTPUT_BUFFER, NMPWAIT_USE_DEFAULT_WAIT, &sa);
+    ExitOnNullWithLastError((hOutTemp && (hOutTemp != INVALID_HANDLE_VALUE)), hr, "Failed to create named pipe for stdout reader");
```

Comment:
Please use the correct exit macros throughout. For example, this would be `ExitOnInvalidHandleWithLastError`.